tests/test_preprocess.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+from pathlib import Path
+import shutil
+import os
+from . import helper
+from silnlp.nmt.config import load_config
+
+
+exp_truth_dir = Path(__file__).parent / "MT/Experiment_Truth"
+
+exp_dir = Path(__file__).parent / "MT/Experiments"
+exp_subdirs = [folder for folder in exp_truth_dir.glob("*/")]
+
+
+@pytest.mark.parametrize("exp_folder", exp_subdirs)
+def test_preprocess(exp_folder):
+    exp_truth_path = os.path.join(exp_truth_dir, exp_folder)
+    config_file = os.path.join(exp_truth_path, "config.yml")
+    assert os.path.isfile(config_file), "The configuraiton file config.yml does not exist for " + exp_folder.name
+    experiment_path = exp_dir / exp_folder.name
+    shutil.rmtree(experiment_path, ignore_errors=True)
```

Comment:
See my suggestion above regarding the use of `tempfile.TemporaryDirectory`.

Reply:
I tried your suggestion - the main issue is that MT_SCRIPTURE_DIR is pre-set in the extract corpora file and is not able to be modified.  To do this, the code would have to be refractored in some way.  One way could be to create a class of "Environment" that is passed into these various functions.  This would involve around 90 modifications.  It is possible - let me try.

Reply:
So, I updated environment to be able to be modifiable (which was needed).  There is still an issue with temp directory.  Often, after a test is run, I want to look at the artifacts to see what changed.  It is much easier to see that in a predictable folder.  I made a "temp_experiments" and "temp_scripture" folder.  If there is a better way of doing it, let me know.

Reply:
Can you not pass the temp directory path to `compare_folders`?

Reply:
I can, but often if there is something "wrong" or "different", we will want to look at it in Beyond Compare, etc.  if it is a temp folder, it would be challenging to find the artifacts made to do the comparison (as happens often).  I am fine putting it anywhere as long as it is deterministic and named.